docs/proposals/data-cache.md

```diff
@@ -0,0 +1,160 @@
+
+Therefore, using Fluid to provide data cache acceleration for the dataset in the KubeDL job is an effective and feasible solution.
+
+### Goals
+
+- Enable KudeDL to transparently create dataset and cache runtime through Fluid (deployed in the same cluster) interface, and provide the generated pvc to the job to complete the cache acceleration of the dataset required by the training job.
+- Provides users with an easy-to-use API and allows them to implement basic parameter configuration as required.
+
+## Use case
+
+User submit deep learning jobs in a Kubernetes cluster with Fluid and KubeDL deployed, and :
+
+1. User wants to specify that the dataset under a certain file path can be cached, so that the time spent on data I/O can be reduced, and the user does not need to know how the dataset is cached.
+
+2. User hopes to be able to specify parameters such as the cache path and the medium used by the Fluid, so that the use of storage resources can be more flexibly allocated according to the needs of users.
+
+3. User hopes to simplify the usage as much as possible, and it is better not to create yaml files other than job.yaml file, so as to reduce the mental burden of users.
+
+## Proposal
+
+Add a new feature, data cache, which will:
+1. Add an data cache field to the job manifest of KubeDL, which will be provided to the user to specify the parameters for the data cache that they will use to flexibly configure how their data will be cached. The parameters will be discussed below.
+2. Add a handler which primary purpose is to parse parameters about the data cache in user-submitted jobs and provide default values for some optional fields.
+3. Add a checker that checks to see if available fluids are deployed within the cluster and, if not, prompts the user that data cache is not available.
+4. Add a client that interacts with the api server. After the parameters related to the data cache are resolved, the parameters are submitted to the fluid in the cluster through the api server and the dataset and create the dataset and cache runtime (usually alluxio runtime), which are apis provided by the fluid.
+5. Add a watch feature that will use client to monitor if a new pv&pvc pair are created in the cluster (if dataset and runtime are successfully created, fluid will bind them and create a pair of pv and pvc).
+6. Add a Injector which injects the pvc created by fluid into the container. This allows jobs to access files in the cached path and enjoy data acceleration.
+
+## Implementation
+
+### Sequence Diagram
+
+FYR. The final implementation may be slightly different from this diagram.
+
+![](../img/data_cache_sequence.png)
+The general process is as follows:
+
+1. The user submits a yaml file describing the job whose field contains the parameters of the data cache, which means that the job will use the feature of the data cache.
```

Comment:
Since we can not guarantee that Fluid is installed (and functioning correctly) in the cluster,  data caching should only be a best-effort protocol provided to users (job submitters). In that case,  we'd better show the caching status (enabled or not, in which path)  in the job status?